eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1408 @@
+
+  if (whatsapp == NULL)
+    return NULL;
+
+  markup = g_strdup_printf ("<div id=\"default-share-actions\" style=\"visibility: hidden;\">%s%s%s</div>",
+                            facebook,
+                            twitter,
+                            whatsapp);
+  return g_variant_new_string (markup);
+}
+
+
+/**
+ * eknr_renderer_render_wrapper:
+ * @renderer: An #EknrRenderer
+ * @content: The body content
+ * @ekn_id: The EKN ID of the enclosing article
+ * @title: Article title
+ * @published: When the article was published
+ * @authors: (element-type utf8): The authors of the article
+ * @license: The article license
+ * @source: The article's source type
+ * @source_name: Translated name of the article's source
+ * @original_uri: Where this articel can be found on the web
+ * @tags: (element-type utf8): The article tags
+ * @set_map: (element-type utf8 EknrBoxedSetInfo): A mapping from set names to sets
+ * @outgoing_links: (element-type utf8): The outgoing links from the article
+ * @determination_func: (scope call): A function used to determine if a link is
+ *                                    accessible offline, or if the user needs
+ *                                    to go online.
+ * @user_data: (closure): Closure for @determination_func
+ * @custom_css_files: (element-type utf8): Any custom css files to include
+ * @override_css_files: (element-type utf8): Override css files
+ * @error: A #GError
+ *
+ * Render the content and return the rendered content.;
+ *
+ * Returns: (transfer full): A string of rendered HTML or %NULL on error.
+ */
+gchar *
+eknr_renderer_render_wrapper (EknrRenderer                       *renderer,
+                              const gchar                        *content,
+                              const gchar                        *ekn_id,
+                              const gchar                        *title,
+                              const gchar                        *published,
+                              GStrv                               authors,
+                              const gchar                        *license,
+                              const gchar                        *source,
+                              const gchar                        *source_name,
+                              const gchar                        *original_uri,
+                              GStrv                               tags,
+                              GHashTable                         *set_map,
+                              GStrv                               outgoing_links,
+                              EknrOutgoingLinkDeterminationFunc   determination_func,
+                              gpointer                            user_data,
+                              GStrv                               custom_css_files,
+                              GStrv                               override_css_files,
```

Comment:
Having to have `GStrv` here in the arguments and then cast it away later on in the function is *really* annoying, but I couldn't get gi-scanner to co-operate with me otherwise (its was not treating `const char * const *` as an array, but it works fine with `GStrv`).